lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -1547,6 +1554,53 @@ void PreCheckExpression::resolveKeyPathExpr(KeyPathExpr *KPE) {
   KPE->resolveComponents(TC.Context, components);
 }
 
+Expr *PreCheckExpression::simplifyInitWithLiteral(Expr *E) {
+  auto *call = dyn_cast<CallExpr>(E);
+  if (!call || call->getNumArguments() != 1)
+    return nullptr;
+
+  auto *typeExpr = dyn_cast<TypeExpr>(call->getFn());
+  if (!typeExpr)
+    return nullptr;
+
+  auto *argExpr = call->getArg()->getSemanticsProvidingExpr();
+  auto *number = dyn_cast<NumberLiteralExpr>(argExpr);
```

Comment:
Yes, I just tried with `NumberLiteralExpr` because I ran into problem with failable initializer in `swiftpm` clashing with `ExpressibleByStringLiteral` but still wanted to get some performance numbers, and `NumberLiteralExpr` since to be the biggest pain point for stdlib folks anyway.